src/lib/UniversalDiscovery/Event/Subscriber/ObjectRelationAllowedContentTypes.php

```diff
@@ -43,10 +43,10 @@ public function onUdwConfigResolve(ConfigResolveEvent $event): void
         }
 
         $config['content_on_the_fly']['allowed_content_types'] = array_unique(
-            array_merge(
+            array_values(array_intersect(
```

Comment:
Be aware, that empty `$config['content_on_the_fly']['allowed_content_types']` array resolves as "Every Content Type is allowed". `array_intersect` will result in empty array in that case, hence will not filter the Content Types by field definition.

We'll need early return of `$context['allowed_content_types']` in case empty configuration array so we don't do intersect at all.

Reply:
Ok, made a change to handle this scenario